src/scripts/main.js

```diff
@@ -1,3 +1,31 @@
 'use strict';
 
-// write code here
+function wrapHeaderIntoSpan(parentElement) {
+  Array.from(parentElement).forEach((child) => {
+    if (child.querySelector('ul')) {
+      const span = document.createElement('span');
+      const childText = child.firstChild;
+
+      span.textContent = childText.textContent.trim();
+      child.replaceChild(span, childText);
+
```

Comment:
You assume that `child.firstChild` contains the text node, but there might be cases where it doesn't. It's better to check if `firstChild `is a text node before manipulating it.
```suggestion
      const childText = child.firstChild;

     if (childText.nodeType === Node.TEXT_NODE) {
       span.textContent = childText.textContent.trim();
       child.replaceChild(span, childText);
    }

```